src/utils/asyncFlatten.ts

```diff
@@ -0,0 +1,6 @@
+export async function asyncFlatten<T>(array: T[]): Promise<T[]> {
+	do {
+		array = (await Promise.all(array)).flat(Infinity) as any;
+	} while (array.some((v: any) => v?.then));
+	return array;
+}
```

Comment:
I wonder it would not make sense to integrate the `filter(Boolean)` here as well so you do not need the optional chaining in `v?.then`? But then maybe this should just become `normalizePluginOption`?

Reply:
This function is copied from [Vite](https://github.com/vitejs/vite/blob/b2c0ee04d4db4a0ef5a084c50f49782c5f88587c/packages/vite/src/node/utils.ts#L1127-L1132).

I think we just keep the function simple and keep the falsy value (it's not referenced by now, but in the future, it could be used in other places). `filter(Boolean)` is done in `normalizePluginOption`.